crates/fuel-core/src/state.rs

```diff
@@ -87,6 +89,12 @@ pub enum WriteOperation {
 }
 
 pub trait TransactableStorage: BatchOperations + Debug + Send + Sync {
+    fn checkpoint(&self) -> DatabaseResult<Database> {
+        Err(DatabaseError::Other(anyhow::anyhow!(
```

Comment:
I think it would be better to add an error variant for `Unsupported` or whatever. I think we should get out of the habit of using `anyhow`.

Reply:
Depends on the situation, in the case of checkpoints it's only used in benchmarking / test contexts and we don't do any special logic based on the error variant other than bailing from the benchmark.

https://www.lpalmieri.com/posts/error-handling-rust/#anyhow-or-thiserror